src/sinks/loki.rs

```diff
@@ -289,11 +289,35 @@ async fn healthcheck(config: LokiConfig, client: HttpClient) -> crate::Result<()
 
     let res = client.send(req).await?;
 
-    if res.status() != http::StatusCode::OK {
-        return Err(format!("A non-successful status returned: {}", res.status()).into());
+    let status = match fetch_status("ready", &config, &client).await? {
+        // Issue https://github.com/timberio/vector/issues/6463
+        http::StatusCode::NOT_FOUND => {
+            warn!("Endpoint `/ready` not found. Retrying healthcheck with top level query.");
+            fetch_status("", &config, &client).await?
+        }
+        status => status,
+    };
+
+    match status {
+        http::StatusCode::OK => Ok(()),
+        _ => Err(format!("A non-successful status returned: {}", res.status()).into()),
+    }
+}
+
+async fn fetch_status(
+    endpoint: &str,
+    config: &LokiConfig,
+    client: &HttpClient,
+) -> crate::Result<http::StatusCode> {
+    let uri = format!("{}{}", config.endpoint.uri, endpoint);
+
+    let mut req = http::Request::get(uri).body(hyper::Body::empty()).unwrap();
```

Comment:
```suggestion
    let mut req = http::Request::get(uri).body(hyper::Body::empty()).expect("building request never fails");
```

Nit since `unwraps()` always stick out to me.